war/src/main/java/io/lumeer/engine/rest/OrganizationService.java

```diff
@@ -0,0 +1,389 @@
+      if (organizationId == null) {
+         throw new IllegalArgumentException();
+      }
+      return organizationFacade.readOrganizationUsers(organizationId);
+   }
+
+   /**
+    * @param organizationId organization id
+    * @param userName user name
+    * @param userRoles list of user roles
+    * @throws UserAlreadyExistsException when user with given name already exists
+    */
+   @POST
+   @Path("/id/{organizationId}/users/{userName}") // TODO: aku adresu?
+   @Consumes(MediaType.APPLICATION_JSON)
+   public void addUserToOrganization(final @PathParam("organizationId") String organizationId, final @PathParam("userName") String userName, final List<String> userRoles) throws UserAlreadyExistsException {
+      if (organizationId == null || userName == null) {
+         throw new IllegalArgumentException();
+      }
+      if (userRoles == null) {
+         organizationFacade.addUserToOrganization(organizationId, userName);
+      } else {
+         organizationFacade.addUserToOrganization(organizationId, userName, userRoles);
+      }
+   }
+
+   /**
+    * @param organizationId organization id
+    * @param userName user name
+    */
+   @DELETE
+   @Path("/id/{organizationId}/users/{userName}")
+   public void removeUserFromOrganization(final @PathParam("organizationId") String organizationId, final @PathParam("userName") String userName) {
+      if (organizationId == null || userName == null) {
+         throw new IllegalArgumentException();
+      }
+      organizationFacade.removeUserFromOrganization(organizationId, userName);
+   }
+
+   /**
+    * @param organizationId organization id
+    * @param userName user name
+    * @return list of roles for given user
+    */
+   @GET
+   @Path("/id/{organizationId}/users/{userName}/roles")
+   @Produces(MediaType.APPLICATION_JSON)
+   public List<String> readUserRoles(final @PathParam("organizationId") String organizationId, final @PathParam("userName") String userName) {
+      if (organizationId == null || userName == null) {
+         throw new IllegalArgumentException();
+      }
+      return organizationFacade.readUserRoles(organizationId, userName);
+   }
+
+   /**
+    * @param organizationId organization id
+    * @param userName user name
+    * @param userRoles list of roles to add
+    */
+   @POST
+   @Path("/id/{organizationId}/users/{userName}/roles")
+   @Consumes(MediaType.APPLICATION_JSON)
+   public void addRolesToUser(final @PathParam("organizationId") String organizationId, final @PathParam("userName") String userName, final List<String> userRoles) {
+      if (organizationId == null || userName == null) {
+         throw new IllegalArgumentException();
+      }
+      organizationFacade.addRolesToUser(organizationId, userName, userRoles);
+   }
+
+   /**
+    * @param organizationId organization id
+    * @param userName user name
+    * @param userRoles list of roles
+    */
+   @PUT
```

Comment:
But having POST for adding roles and PUT for removing them is very confusing. We should rather have a single PUT method which will be idempotent and will override all roles. But the underlying facade methods might need to be changed as well to support this.

The other solution might be adding and removing only a single roles. In that case it can be specified in the path.

But we do not know what exactly will be needed in the UI. Maybe both solutions can be implemented if possible. What do you think?